File: exams/villagomez/u2/Q71-100 - DOMENICA NICOLE VILLAGOMEZ FREIRE/Q71-100/EFriends/src/main/java/utils/MongoDB.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package utils;

import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoDatabase;

/**
 *
 * @author G406
 */
public class MongoDB {

    private static MongoClient mongoClient;
    private static MongoDatabase database;

    /*private MongoDBConnection() {
    }

    public static synchronized MongoDatabase getDatabase() {
        if (database == null) {
            String uri = "mongodb+srv://villagomez:villagomez@cluster0.hdylr9r.mongodb.net/";
            mongoClient = MongoClients.create(uri);
            database = mongoClient.getDatabase(
        
        "poo);
        }
        return database;
    }

    public static void close() {
        if (mongoClient != null) {
            mongoClient.close();
            mongoClient = null;
            database = null;
        }
    }*/
}

File: quizz/segarra/u3/Quizz12/utils/mongoDB.mjs
import { MongoClient } from 'mongodb';

const uri = 'mongodb+srv://segarra:segarra@cluster0.b2q6ac3.mongodb.net/';
const client = new MongoClient(uri);

export async function saveStoreToDatabase(store) {
    try {
        await client.connect();
        const database = client.db('Q6170');
        const stores = database.collection('Stores');
        await stores.insertOne(store);
    } finally {
        await client.close();
    }
}

